cmd/client/main.go

```diff
@@ -211,6 +223,84 @@ func (c *Client) run(o *GrpcProxyClientOptions) error {
 }
 
 func (c *Client) getDialer(o *GrpcProxyClientOptions) (func(ctx context.Context, network, addr string) (net.Conn, error), error) {
+	if o.proxyUdsName != "" {
+		return c.getUDSDialer(o)
+	}
+	return c.getMTLSDialer(o)
+}
+
+func (c *Client) getUDSDialer(o *GrpcProxyClientOptions) (func(ctx context.Context, network, addr string) (net.Conn, error), error) {
+	var proxyConn net.Conn
+	var err error
+
+	// Setup signal handler
+	ch := make(chan os.Signal, 1)
+	signal.Notify(ch)
+
+	go func() {
+		<-ch
+		err := proxyConn.Close()
+		klog.Infof("connection closed: %v", err)
+	}()
+
+	switch o.mode {
+	case "grpc":
+		dialOption := grpc.WithDialer(func(string, time.Duration) (net.Conn, error) {
+			// Ignoring addr and timeout arguments:
+			// addr - comes from the closure
+			// timeout - is ignored since we are connecting in a non-blocking configuration
```

Comment:
`client.CreateGrpcTunnel` is blocking, right?

Reply:
Missed the recv(). Fixed the comment. Leaving the 0 timeout.